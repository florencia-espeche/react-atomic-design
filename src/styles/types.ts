import { FlattenSimpleInterpolation } from 'styled-components';

export type Theme = {
    name: string
    colors: {
        primary: string
        secondary: string
        neutral: {
            white: string
            light: {
                300: string
                500: string
            }
            dark: string
            soft: string
            medium: string
        }
        suport: {
            success: string
            warning: string
            error: string
            successLight: string
            warningLight: string
            errorLight: string
        }
        blue: {
            light: string
            medium: string
            dark: string
        }
        gray: {
            light: string
            medium: string
            dark: string
        }
    }
    fontTokens: {
        [key: string]: FlattenSimpleInterpolation
    }
    borderRadius: string
    fontWeight: {
        light: number
        regular: number
        medium: number
        bold: number
        black: number
    }
}
