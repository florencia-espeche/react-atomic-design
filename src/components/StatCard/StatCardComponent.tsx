import React from 'react'
import { useTheme } from '../../hooks/useTheme'

import {
    Container,
    Title,
    Subtitle,
    Quantity,
    ProgressContainer,
    Progress
} from './StatCardComponent.styles'
import { StatCardProps } from './StatCardComponent.types'

const StatCard = ({
    title,
    variation = 'primary',
    subtitle,
    quantity,
    total,
    children
}: StatCardProps) => {
    const theme = useTheme()
    const percentage = Math.min((quantity / total) * 100, 100)

    return (
        <Container variation={variation}>
            <Title theme={theme}>{title}</Title>
            <Subtitle theme={theme}>{subtitle}</Subtitle>
            <Quantity theme={theme}>{`${quantity}/${total}`}</Quantity>
            <ProgressContainer theme={theme}>
                <Progress theme={theme} percentage={percentage} />
            </ProgressContainer>
            {children}
        </Container>
    )
}

export default StatCard
