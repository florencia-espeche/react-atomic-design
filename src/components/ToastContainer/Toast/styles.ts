import styled, { css } from 'styled-components'
import { animated } from 'react-spring'
const ToastTypes = {
    success: css`
        background-color: ${props => props.theme.colors.suport.successLight};
    `,
    warning: css`
        background-color: ${props => props.theme.colors.suport.warningLight};
    `,
    error: css`
        background-color: ${props => props.theme.colors.suport.errorLight};
    `
}

interface ContainerProps {
    toastType: 'info' | 'success' | 'warning' | 'error'
}

export const Container = styled(animated.div)<ContainerProps>`
    padding: 0 18px;
    position: relative;
    border-radius: ${props => props.theme.borderRadius};
    box-shadow: 2px 2px 8px rgba(0, 0, 0, 0.2);
    width: auto;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 48px;
    min-width: 327px;
    ${props => ToastTypes[props.toastType]};

    & + div {
        margin-top: 8px;
    }
    > div {
        display: flex;
        align-items: center;
        justify-content: center;
        > svg {
            margin-right: 10px;
        }
        p {
            ${props => props.theme.fontTokens['100']}
        }
    }

    button {
        color: red;
        border: none;
        background: none;
        cursor: pointer;
        outline: none;
        height: 13px;
        width: 13px;

        svg {
            height: 13px;
            width: 13px;
        }
    }
`
