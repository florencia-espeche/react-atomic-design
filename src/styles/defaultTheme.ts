import { css } from 'styled-components'
import { Theme } from './types'

export const defaultTheme: Theme = {
    name: 'default',
    colors: {
        primary: '#2B52DD',
        secondary: '#004282 ',
        neutral: {
            white: '#FFF',
            light: {
                '300': '#F4F6FC',
                '500': '#C9CAD2'
            },
            soft: '#A7ADBF',
            medium: '#6D7384',
            dark: '#2B2C32'
        },
        suport: {
            success: '#3EB96F',
            warning: '#FBC436',
            error: '#E73A33',
            successLight: '#D8F1E2',
            warningLight: '#FEF3D7',
            errorLight: '#FAD8D6'
        },
        blue: {
            light: '#CFDDF5',
            medium: '#173EC5',
            dark: '#163161'
        },
        gray: {
            light: '#C9CAD2',
            medium: '#EBECF4',
            dark: '#6D6E75'
        }
    },
    borderRadius: '4px',
    fontTokens: {
        '50': css`
            font-size: 11px;
            line-height: 13px;
        `,
        '75': css`
            font-size: 12px;
            line-height: 15px;
        `,
        '100': css`
            font-size: 14px;
            line-height: 17px;
        `,
        '200': css`
            font-size: 16px;
            line-height: 19px;
        `,
        '300': css`
            font-size: 18px;
            line-height: 22px;
        `,
        '400': css`
            font-size: 20px;
            line-height: 24px;
        `,
        '500': css`
            font-size: 22px;
            line-height: 27px;
        `,
        '600': css`
            font-size: 24px;
            line-height: 31px;
        `,
        '700': css`
            font-size: 28px;
            line-height: 34px;
        `,
        '800': css`
            font-size: 32px;
            line-height: 39px;
        `,
        '900': css`
            font-size: 36px;
            line-height: 44px;
        `,
        '1000': css`
            font-size: 40px;
            line-height: 49px;
        `,
        '1100': css`
            font-size: 45px;
            line-height: 55px;
        `,
        '1200': css`
            font-size: 50px;
            line-height: 62px;
        `,
        '1300': css`
            font-size: 60px;
            line-height: 70px;
        `
    },
    fontWeight: {
        light: 300,
        regular: 400,
        medium: 500,
        bold: 700,
        black: 900
    }
}
