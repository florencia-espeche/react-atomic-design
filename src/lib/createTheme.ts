import { defaultTheme } from '../styles/defaultTheme'
import { Theme } from '../styles/types'

export const createTheme = <T>(theme: Theme & T) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },
        borderRadius: theme.borderRadius || defaultTheme.borderRadius,
        fontTokens: {
            ...defaultTheme.fontTokens,
            ...theme.fontTokens
        },
        fontWeight: theme.fontWeight || defaultTheme.fontWeight,
        name: theme.name || defaultTheme.name
    }
}
