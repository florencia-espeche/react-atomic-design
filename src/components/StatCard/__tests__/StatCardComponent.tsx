import { render } from '../../../lib/test-utils'

import StatCardComponent from '../StatCardComponent'
import { Link } from '../StatCardComponent.styles'

const args = {
    title: 'Title',
    subtitle: 'Subtitle',
    quantity: 8,
    total: 10
}

const Children = () => (
    <div>
        <Link>Children</Link>
    </div>
)

describe('<StatCardComponent />', () => {
    it('Should render the primary variation by default', () => {
        const { container } = render(<StatCardComponent {...args} />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should render the secondary variation', () => {
        const { container } = render(
            <StatCardComponent {...args} variation="secondary" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should render with children', () => {
        const { container } = render(
            <StatCardComponent {...args} children={<Children />} />
        )

        expect(container.firstChild).toMatchSnapshot()
    })
})
