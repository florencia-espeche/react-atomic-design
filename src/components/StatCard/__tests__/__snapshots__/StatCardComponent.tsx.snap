// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatCardComponent /> Should render the primary variation by default 1`] = `
<div
  class="sc-bdfBQB GIkpO"
>
  <span
    class="sc-gsTEea eskIjD"
  >
    Title
  </span>
  <span
    class="sc-dlfnuX irNcAN"
  >
    Subtitle
  </span>
  <span
    class="sc-hKgJUU hxnvbo"
  >
    8/10
  </span>
  <div
    class="sc-eCstlR bMpgvh"
  >
    <div
      class="sc-jSgvzq igPcSV"
    />
  </div>
</div>
`;

exports[`<StatCardComponent /> Should render the secondary variation 1`] = `
<div
  class="sc-bdfBQB hDZwdz"
>
  <span
    class="sc-gsTEea eskIjD"
  >
    Title
  </span>
  <span
    class="sc-dlfnuX irNcAN"
  >
    Subtitle
  </span>
  <span
    class="sc-hKgJUU hxnvbo"
  >
    8/10
  </span>
  <div
    class="sc-eCstlR bMpgvh"
  >
    <div
      class="sc-jSgvzq igPcSV"
    />
  </div>
</div>
`;

exports[`<StatCardComponent /> Should render with children 1`] = `
<div
  class="sc-bdfBQB GIkpO"
>
  <span
    class="sc-gsTEea eskIjD"
  >
    Title
  </span>
  <span
    class="sc-dlfnuX irNcAN"
  >
    Subtitle
  </span>
  <span
    class="sc-hKgJUU hxnvbo"
  >
    8/10
  </span>
  <div
    class="sc-eCstlR bMpgvh"
  >
    <div
      class="sc-jSgvzq igPcSV"
    />
  </div>
  <div>
    <a
      class="sc-gKseQn hMRgoU"
    >
      Children
    </a>
  </div>
</div>
`;
